### PR TITLE
Tab summaries endpoint

### PR DESCRIPTION
**Pull request by sultan-duisenbay**
Extend API by adding `/tab-summaries` endpoint

### CODE REVIEW

File: pkg/api/v1/summary_test.go
Comment (chases2) [inline pkg/api/v1/summary_test.go:51]: This test returns an empty tab summary for an empty summary file. How would your code react if the file actually wasn't there? (ex: a tab is added to config and requested before the summarizer has summarized it for the first time)

File: pkg/api/v1/summary.go
Comment (chases2) [inline pkg/api/v1/summary.go:111]: Not clear from this TODO what issue you're talking about here. Do you mean just checking if summary is nil AND error is nil?

File: pb/api/v1/data.proto
Comment (michelle192837) [inline pb/api/v1/data.proto:175]: nit: Not aligned with other fields?

File: pb/api/v1/data.proto
Comment (michelle192837) [inline pb/api/v1/data.proto:194]: nit: Maybe we want to do latest_passing_build instead for clarity?

**Review by chases2 — COMMENTED** [review]
This is an excellent start! A couple of comments around tests and TODOs, but the implementation looks great overall.

/lint

File: pkg/api/v1/summary_test.go
Comment (michelle192837) [inline pkg/api/v1/summary_test.go:2]: nit: 2023 right?

**Review by michelle192837 — COMMENTED** [review]
Broadly looks good, have a couple nits in addition to Sean's feedback. Thanks Sultan! ^^

File: pkg/api/v1/config_cache.go
Comment (sultan-duisenbay) [reply pkg/api/v1/config_cache.go]: Done

File: pkg/summarizer/summary.go
Comment (sultan-duisenbay) [reply pkg/summarizer/summary.go:383]: Done

File: pkg/summarizer/summary.go
Comment (sultan-duisenbay) [reply pkg/summarizer/summary.go:404]: Done

File: pkg/api/v1/summary.go
Comment (sultan-duisenbay) [reply pkg/api/v1/summary.go]: Sounds good! Done

File: pkg/api/v1/config_cache.go
Comment (sultan-duisenbay) [reply pkg/api/v1/config_cache.go]: Good catch, done!

File: pkg/api/v1/summary_test.go
Comment (sultan-duisenbay) [reply pkg/api/v1/summary_test.go]: if summary doesn't exist yet, the error is returned. This follows the same path as ListHeader call. I added the test case to explicitly check that

File: pkg/api/v1/summary.go
Comment (sultan-duisenbay) [reply pkg/api/v1/summary.go]: Yes, I added the check now.

File: pb/api/v1/data.proto
Comment (sultan-duisenbay) [reply pb/api/v1/data.proto:175]: That's funny, since it was aligned properly in my branch :) I will try to push some changes and let's see if this gets sorted out.

File: pb/api/v1/data.proto
Comment (sultan-duisenbay) [reply pb/api/v1/data.proto]: Done!

File: pkg/api/v1/summary_test.go
Comment (sultan-duisenbay) [reply pkg/api/v1/summary_test.go]: those 2 years never happened, we are still in 2021 lol.

on a more serious note, done

File: pkg/api/v1/summary_test.go
Comment (michelle192837) [inline pkg/api/v1/summary_test.go:68]: nit: Just for clarity on this test case, think you can make this an empty map.

**Review by michelle192837 — APPROVED** [review]
/hold

LGTM, holding if Sean still wants to review

File: pkg/api/v1/summary.go
Comment (chases2) [inline pkg/api/v1/summary.go:104]: nit: It looks like tabSummary.LastRunTimestamp is a float. This intrinsically rounds the timestamp to the nearest second, which is probably fine: don't feel like you need to do this in this PR.

More precisely, though, you may want to convert the fractional part of the float into nanos. Example 5 of this kind of illustrates what I mean: https://developers.google.com/protocol-buffers/docs/reference/google.protobuf#google.protobuf.Timestamp

File: pb/api/v1/data.proto
Comment (chases2) [reply pb/api/v1/data.proto:175]: Still looks borked on GitHub. Make sure you (more likely, your text editor) aren't mixing tabs and spaces.

File: pkg/api/v1/summary_test.go
Comment (chases2) [inline pkg/api/v1/summary_test.go:68]: This dashboard name doesn't match the config, and shouldn't be read from, right?

Please either do not include it or name it "summary-unrelateddashboard" or whatever if that's what you're meaning.

**Review by chases2 — APPROVED** [review]
A couple of nits left in comments, but looks good overall. I also approve this

File: pkg/api/v1/summary_test.go
Comment (chases2) [reply pkg/api/v1/summary_test.go:68]: +1, that would be clearer

Comment (sultan-duisenbay) [reply pkg/api/v1/summary_test.go:68]: 100% agree, done!

File: pkg/api/v1/summary_test.go
Comment (sultan-duisenbay) [reply pkg/api/v1/summary_test.go:68]: Addressed this above. Michelle also pointed it out - it's empty now.

File: pkg/api/v1/summary.go
Comment (sultan-duisenbay) [reply pkg/api/v1/summary.go:104]: Sounds good, I will add this as a TODO. Thanks!

**Pull request merged**
